packages/core/src/shared/telemetry/util.ts

```diff
@@ -90,6 +91,37 @@ export function convertLegacy(value: unknown): boolean {
     }
 }
 
+/**
+ * Returns an identifier that uniquely identifies a single application
+ * instance/window of a specific IDE. I.e if I have multiple VS Code
+ * windows open each one will have a unique session ID. This session ID
+ * can be used in conjunction with the client ID to differntiate between
+ * different VS Code windows on a users machine.
+ *
+ * ### Rules:
+ *
+ * - On startup of a new application instance, a new session ID must be created.
+ *   - This identifier must be a `UUID`, it is enforced by the telemetry service.
+ * - The session ID must be different from all other IDs on the machine
+ * - A session ID exists until the application instance is terminated.
+ *   It should never be used again after termination.
+ * - All extensions on the same application instance should return the same session ID.
```

Comment:
This might be an interesting, but is it actually needed? Perhaps this works by accident because of the way things work in vscode, but it might be difficult in other IDEs/systems? Does it matter either way (if not, probably shouldn't be a requirement)?

Reply:
I was thinking that if the Amazon Q and Toolkit extension are in the same window we want to know that since they interact with eachother (eg connection sharing).

This isn't enforced on the service side, so if other IDEs are not able to, then they can use different UUIDs. But then things won't be consistent across IDEs though.

Reply:
> if the Amazon Q and Toolkit extension are in the same window we want to know

Could be useful. (Though it's an edge case since client-id will usually tell us that.)

Reply:
ClientID does not tell us this since it is the same across all VS Code windows on a machine. This was the original issue, if I had multiple VS Code windows with Amazon Q extension, and they were all sending telemetry, I could not differentiate between them. SessionID solves this.